dask/dataframe/io.py

```diff
@@ -665,8 +673,10 @@ def from_delayed(dfs, metadata=None, divisions=None, columns=None,
 
     if divisions is None:
         divisions = [None] * (len(dfs) + 1)
+    if metadata is None:
+        metadata = dfs[0].compute()
```

Comment:
The constructor of frame objects trim this down, so this happens in the next line or so anyway.